_source/_posts/2020-12-09-dotnet-cloud-host-publish.md

```diff
@@ -0,0 +1,325 @@
+
+*Login redirect URIs*: `https://<your_app_name>.azurewebsites.net/authorization-code/callback`
+
+*Logout redirect URIs*: `https://<your_app_name>.azurewebsites.net/signout/callback`
+
+Now you should be able to log in and log out in your web application using your Okta credentials.
+
+## Deploying a .NET Core 3.1 App to AWS Beanstalk
+
+Next, we will go through the deployment process for AWS.
+
+### Creating the deployment package
+
+Open a terminal or  command prompt, go to the project folder and run:
+
+```sh
+dotnet publish -o my-app
+```
+
+Compress the newly created `my-app` folder to a zip archive. This will be our deployment package
+
+### Deploying the package
+
+Visit the AWS [Elastic Beanstalk](https://console.aws.amazon.com/elasticbeanstalk) Console.
+
+Go ahead and *Create Application*:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image9.png alt:"" width:"800" %}{: .center-image }
+
+Give it a *Name* and carefully select the Platform as follows:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image10.png alt:"" width:"800" %}{: .center-image }
+
+We'll upload the deployment package by hand. So, set Application Code to *Upload your code*. Now press *Choose file* and upload the previously created .zip file.
+
+Once the upload has finished, hit *Create Application*. You might have to wait a few minutes for the deployment to complete.
+
+### Configuring Okta for the AWS Hosted App
+
+We need to add our application's URL to Okta to be recognized as a redirect URL.
+
+Going back to the Okta Dashboard Applications page. Select the App you created in the first part.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image11.png alt:"" width:"800" %}{: .center-image }
+
+On the General tab hit Edit by the General Settings section:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image12.png alt:"" width:"800" %}{: .center-image }
+
+Set the following URLs:
+*Login redirect URIs*: `http://<your_app_url>/authorization-code/callback`
+e.g.: `http://oktaexample-env.eba-evyiprzp.us-east-1.elasticbeanstalk.com/authorization-code/callback`
+
+*Logout redirect URIs*: `http://<your_app_url>/signout/callback`
+e.g.: `http://oktaexample-env.eba-evyiprzp.us-east-1.elasticbeanstalk.com/signout/callback`
+
+Now you should be able to log in and log out in your web application using your Okta credentials.
+
+### Troubleshooting
+
+If you run into a Correlation Error after logging in, you'll need to manually set the `SameSite` cookie attribute to `None`, and enable SSL (HTTPS) on your server. I have a detailed article and a guide coming soon on how to set up a Linux server to support Okta login in .NET Core 3.1 applications - be on the lookout for that! To get an idea of the basics, check out my post on how to [Install .NET Core Apps on Linux in 5 Minutes](https://developer.okta.com/blog/2020/11/25/how-to-install-dotnetcore-on-linux).
+
+## Deploying a .NET Core 3.1 App to Google App Engine
+
+### Preparing Google Cloud App Engine
```

Comment:
I think it looks funny to have two headers in a row.

Reply:
I do too but it passed editorial and I guess that doesn't matter anymore?